### PR TITLE
Code cleanup in System.ComponentModel.TypeConverter

### LINKED ISSUE #12160: Code cleanup for System.ComponentModel.TypeConverter
There are a couple of clean-up items for the newly ported code inside System.ComponentModel.TypeConverter. These are captured from PR #12081 
- Remove explicit base() from constructor calls
- Replace explicitly implemented properties with auto-implemented properties when possible
- Simplify members and properties with 1 lines of code to expression bodied members: protected virtual bool CanRaiseEvents => true;
  *Switch over to the ?. operator in C# for null checking:

``` csharp
//From
EventHandler handler = (EventHandler)_events[s_eventDisposed];
if (handler != null) handler(this, EventArgs.Empty);
//To
((EventHandler)_events[s_eventDisposed])?.Invoke(this, EventArgs.Empty);

//From
ISite s = _site;
return s == null ? null : s.Container;
//To
public IContainer Container => _site?.Container;
```
- Switch to string interpolation instead of string concatenation
- Remove unnecessary initialization (ie. Bool foo = false)
- Replace empty arrays with Array.Empty<T>()
- Replace string literals with nameof wherever possible.
- Simplify event handler declarations when no custom code is required
- Introduce readonly modifier for sync objects.
- Replace nongeneric collections with generic collections
- Use foreach instead of manually writing the foreach logic.
- Cache the char[] into a static readonly field (ie. new char[] { '\' })
- Simplify the CultureInfoMapper code:

``` csharp
private static class CultureInfoMapper
{
    // Dictionary of CultureInfo.DisplayName, CultureInfo.Name for cultures that have changed DisplayName over releases.
    // This is to workaround an issue with CultureInfoConverter that serializes DisplayName (fixing it would introduce breaking changes).
    private static readonly Dictionary<string, string> s_cultureInfoNameMap = CreateMap();

    private static Dictionary<string, string> CreateMap()
    {
        const int Count = 274;
        var result = new Dictionary<string, string>(Count)
        {
            { "Afrikaans", "af" },
            ...
        };
        Debug.Assert(result.Count == Count);
        return result;
    }

    public static string GetCultureInfoName(string cultureInfoDisplayName)
    {
        string name;
        return s_cultureInfoNameMap.TryGetValue(cultureInfoDisplayName, out name) ?
            name :
            cultureInfoDisplayName;
    }
}
```

### PR DESCRIPTION
**Pull request by bmeverett**
Resolves #12160 to clean up code in System.ComponentModel.TypeConverter
cc @AlexGhiondea 

- [x] Remove explicit base() from constructor calls
- [x] Replace explicitly implemented properties with auto-implemented properties when possible
- [x] Simplify members and properties with 1 lines of code to expression bodied members
- [x] Switch over to the ?. operator in C# for null checking
- [x] Switch to string interpolation instead of string concatenation
- [x] Remove unnecessary initialization (ie. Bool foo = false)
- [x] Replace empty arrays with Array.Empty()
- [x] Replace string literals with nameof wherever possible.
- [x] Simplify event handler declarations when no custom code is required
- [x] Introduce readonly modifier for sync objects.
- [x] Replace nongeneric collections with generic collections
- [x] Use foreach instead of manually writing the foreach logic.
- [x] Cache the char[] into a static readonly field (ie. new char[] { '' })
- [x] Simplify the CultureInfoMapper code

I created an individual commit for each improvement, didn't know if that would make review any easier.  I tried to search the project and make sure everything was accounted for. This is a decent size project so something may have been missed. Feel free to comment and I'll be glad to fix it.

### CODE REVIEW

**Comment by AlexGhiondea** [comment]
@bmeverett thanks for your contribution! I am going to take a closer look at this next week when I get back in the office!

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/AttributeCollection.cs [hunk 1]
```diff
@@ -107,7 +107,7 @@ public static AttributeCollection FromExisting(AttributeCollection existing, par
             // Now, if we collapsed some attributes, create a new array.
             //
 
-            Attribute[] attributes = null;
+            Attribute[] attributes = Array.Empty<Attribute>();
```
Comment (stephentoub) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/AttributeCollection.cs]: Does this need to be assigned at all?  It's set in both the subsequent if/else branch, so it should be definitely assigned without needing this initializer.

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/ComplexBindingPropertiesAttribute.cs [hunk 2]
```diff
@@ -26,8 +23,8 @@ public sealed class ComplexBindingPropertiesAttribute : Attribute
         /// </summary>
         public ComplexBindingPropertiesAttribute()
         {
-            _dataSource = null;
-            _dataMember = null;
+            DataSource = null;
+            DataMember = null;
```
Comment (stephentoub) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/ComplexBindingPropertiesAttribute.cs]: Are these lines necessary at all?  null is the default value.

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/ComplexBindingPropertiesAttribute.cs [hunk 3]
```diff
@@ -38,8 +35,8 @@ public ComplexBindingPropertiesAttribute()
         /// </summary>
         public ComplexBindingPropertiesAttribute(string dataSource)
         {
-            _dataSource = dataSource;
-            _dataMember = null;
+            DataSource = dataSource;
+            DataMember = null;
```
Comment (stephentoub) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/ComplexBindingPropertiesAttribute.cs]: Is this line necessary?  null is the default value.

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/CultureInfoConverter.cs [hunk 4]
```diff
@@ -111,17 +106,16 @@ public override object ConvertFrom(ITypeDescriptorContext context, CultureInfo c
                     //
                     if (retVal == null)
                     {
-                        ICollection values = GetStandardValues(context);
-                        IEnumerator e = values.GetEnumerator();
-                        while (e.MoveNext())
+                        foreach (var val in GetStandardValues(context))
                         {
-                            CultureInfo info = (CultureInfo)e.Current;
+                            CultureInfo info = (CultureInfo)val;
```
Comment (stephentoub) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/CultureInfoConverter.cs]: This can be changed to just:
```C#
foreach (CultureInfo info in GetStandardValues(context))
```

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/CultureInfoConverter.cs [hunk 5]
```diff
@@ -140,16 +134,16 @@ public override object ConvertFrom(ITypeDescriptorContext context, CultureInfo c
                     if (retVal == null)
                     {
                         text = text.ToLower(CultureInfo.CurrentCulture);
-                        IEnumerator e = _values.GetEnumerator();
-                        while (e.MoveNext())
+                        foreach (var val in _values)
                         {
-                            CultureInfo info = (CultureInfo)e.Current;
+                            CultureInfo info = (CultureInfo)val;
```
Comment (stephentoub) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/CultureInfoConverter.cs]: Ditto

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/CommandID.cs [hunk 6]
```diff
@@ -94,7 +82,7 @@ public virtual Guid Guid
         /// </summary>
         public override string ToString()
         {
-            return _menuGroup.ToString() + " : " + _commandID.ToString(CultureInfo.CurrentCulture);
+            return $"{_menuGroup.ToString()} : {_commandID.ToString(CultureInfo.CurrentCulture)}";
```
Comment (stephentoub) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/CommandID.cs]: Same comment.  In general, we should be careful about using string interpolation in anything other than debug code.  In some cases it may make sense, but string.Concat with string inputs is faster, especially when the number of input strings is <= 4, as there are string.Concat overloads for handling those cases explicitly (and the compiler maps string concatenation to using string.Concat).

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/DesignerOptionService.cs [hunk 7]
```diff
@@ -226,29 +200,29 @@ public PropertyDescriptorCollection Properties
...
                         {
-                            propList = new ArrayList(1);
+                            propList = new List<PropertyDescriptor>(1);
                         }
 
                         EnsurePopulated();
                         foreach (DesignerOptionCollection child in _children)
                         {
-                            propList.AddRange(child.Properties);
+                            propList.AddRange(Enumerable.Cast<PropertyDescriptor>(child.Properties));
```
Comment (stephentoub) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/DesignerOptionService.cs]: This adds expense, at least an allocation, plus additional virtual method calls.  Probably best to revert these changes and stick with using ArrayList, since the input we're dealing with is itself non-generic.

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/DesignerVerb.cs [hunk 8]
```diff
@@ -85,7 +85,7 @@ public string Text
         /// </summary>
         public override string ToString()
         {
-            return Text + " : " + base.ToString();
+            return $"{Text} : {base.ToString()}";
```
Comment (stephentoub) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/DesignerVerb.cs]: Same comment about string interpolation.  I'll stop commenting on these.

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/DesigntimeLicenseContext.cs [hunk 9]
```diff
@@ -175,8 +170,8 @@ public override string GetSavedLicenseKey(Type type, Assembly resourceAssembly)
 
                 if (licenseFile != null)
                 {
-                    Debug.WriteLineIf(s_runtimeLicenseContextSwitch.TraceVerbose, "licenseFile: " + licenseFile.ToString());
-                    Debug.WriteLineIf(s_runtimeLicenseContextSwitch.TraceVerbose, "opening licenses file over URI " + licenseFile.ToString());
+                    Debug.WriteLineIf(s_runtimeLicenseContextSwitch.TraceVerbose, $"licenseFile: {licenseFile.ToString()}");
+                    Debug.WriteLineIf(s_runtimeLicenseContextSwitch.TraceVerbose, $"opening licenses file over URI {licenseFile.ToString()}");
```
Comment (stephentoub) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/DesigntimeLicenseContext.cs]: (Since these are in Debug statements, this usage of string interpolation is fine)

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/Serialization/InstanceDescriptor.cs [hunk 10]
```diff
@@ -33,18 +29,18 @@ public InstanceDescriptor(MemberInfo member, ICollection arguments) : this(membe
...
         {
-            _member = member;
-            _isComplete = isComplete;
+            MemberInfo = member;
+            IsComplete = isComplete;
 
             if (arguments == null)
             {
-                _arguments = new object[0];
+                Arguments = new object[0];
```
Comment (stephentoub) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/Serialization/InstanceDescriptor.cs]: Should this be ```Array.Empty<object>()``` instead of ```new object[0]```?  I'm wondering if there's some reason to care about the reference identity in this case.

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/EnumConverter.cs [hunk 11]
```diff
@@ -16,6 +16,7 @@ namespace System.ComponentModel
     /// </summary>
     public class EnumConverter : TypeConverter
     {
+        private static readonly char[] c_comma = {','};
```
Comment (stephentoub) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/EnumConverter.cs]: Nit: We use ```s_``` rather than ```c_``` as prefixes for static fields.

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/MaskedTextProvider.cs [hunk 12]
```diff
@@ -548,37 +527,17 @@ public object Clone()
...
 
         /// <summary>
         ///       The system password char.
         /// </summary>
-        public static char DefaultPasswordChar
-        {
-            get
-            {
-                // ComCtl32.dll V6 (WindowsXP) provides a nice black circle but we don't want to attempt to simulate it 
-                // here to avoid hard coding values.  MaskedTextBox picks up the right value at run time from comctl32.
```
Comment (stephentoub) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/MaskedTextProvider.cs]: Why was this comment deleted?

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/MemberDescriptor.cs [hunk 13]
```diff
@@ -97,19 +97,19 @@ protected MemberDescriptor(MemberDescriptor oldMemberDescriptor, Attribute[] new
             _displayName = oldMemberDescriptor.DisplayName;
             _nameHash = _name.GetHashCode();
 
-            ArrayList newArray = new ArrayList();
+            List<Attribute> newArray  = new List<Attribute>();
```
Comment (stephentoub) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/MemberDescriptor.cs]: Nit: extra space before the =

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/ReflectTypeDescriptionProvider.cs [hunk 14]
```diff
@@ -228,12 +222,7 @@ public override object CreateInstance(IServiceProvider provider, Type objectType
                 obj = objectType.GetTypeInfo().GetConstructor(argTypes)?.Invoke(args);
             }
 
-            if (obj == null)
-            {
-                obj = Activator.CreateInstance(objectType, args);
-            }
-
-            return obj;
+            return obj ?? (obj = Activator.CreateInstance(objectType, args));
```
Comment (stephentoub) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/ReflectTypeDescriptionProvider.cs]: Why does obj need to be assigned? e.g. why not just:
```C#
return obj ?? Activator.CreateInstance(objectType, args);
```
?

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/TypeDescriptor.cs [hunk 15]
```diff
@@ -498,21 +486,13 @@ public static object CreateInstance(IServiceProvider provider, Type objectType,
...
-            if (instance == null)
+            TypeDescriptionProvider p = provider?.GetService(typeof(TypeDescriptionProvider)) as TypeDescriptionProvider;
+            if (p != null)
             {
-                instance = NodeFor(objectType).CreateInstance(provider, objectType, argTypes, args);
+                instance = p.CreateInstance(provider, objectType, argTypes, args);
             }
 
-            return instance;
+            return instance ?? (instance = NodeFor(objectType).CreateInstance(provider, objectType, argTypes, args));
```
Comment (stephentoub) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/TypeDescriptor.cs]: Why does instance need to be assigned?

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/Serialization/InstanceDescriptor.cs (hunk 10 above)
Comment (bmeverett) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/Serialization/InstanceDescriptor.cs]: Looks like `Array.Empty<object>` would accomplish the same thing.

**Comment by bmeverett** [comment]
@stephentoub Thanks for the feedback! Especially about string interpolation. I will keep that in mind. I will fix up your comments and make another commit.

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/ByteConverter.cs [hunk 16]
```diff
@@ -16,13 +16,7 @@ public class ByteConverter : BaseNumberConverter
...
         /// The Type this converter is targeting (e.g. Int16, UInt32, etc.)
         /// </summary>
-        internal override Type TargetType
-        {
-            get
-            {
-                return typeof(Byte);
-            }
-        }
+        internal override Type TargetType => typeof(Byte);
```
Comment (AlexGhiondea) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/ByteConverter.cs]: Byte -> byte

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/CultureInfoConverter.cs [hunk 17]
```diff
@@ -339,26 +331,12 @@ private static class CultureInfoMapper
...
-                if (s_cultureInfoNameMap == null)
-                {
-                    InitializeCultureInfoMap();
-                }
-
-                if (s_cultureInfoNameMap.ContainsKey(cultureInfoDisplayName))
-                {
-                    return s_cultureInfoNameMap[cultureInfoDisplayName];
-                }
+            private static readonly Dictionary<string, string> s_cultureInfoNameMap = CreateMap();
```
Comment (AlexGhiondea) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/CultureInfoConverter.cs]: Was the removal of `volatile` intentional?

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/DecimalConverter.cs [hunk 18]
```diff
@@ -19,24 +19,12 @@ public class DecimalConverter : BaseNumberConverter
...
         /// The Type this converter is targeting (e.g. Int16, UInt32, etc.)
         /// </summary>
-        internal override Type TargetType
-        {
-            get
-            {
-                return typeof(Decimal);
-            }
-        }
+        internal override Type TargetType => typeof(Decimal);
```
Comment (AlexGhiondea) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/DecimalConverter.cs]: Decimal -> decimal

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/DefaultBindingPropertyAttribute.cs [hunk 19]
```diff
@@ -24,7 +22,7 @@ public sealed class DefaultBindingPropertyAttribute : Attribute
         /// </summary>
         public DefaultBindingPropertyAttribute()
         {
-            _name = null;
+            Name = null;
```
Comment (AlexGhiondea) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/DefaultBindingPropertyAttribute.cs]: This is not required

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/DefaultEventAttribute.cs [hunk 20]
```diff
@@ -33,13 +33,7 @@ public DefaultEventAttribute(string name)
...
         ///    </para>
         /// </summary>
-        public string Name
-        {
-            get
-            {
-                return _name;
-            }
-        }
+        public string Name => _name;
```
Comment (AlexGhiondea) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/DefaultEventAttribute.cs]: Can we remove `_name`?

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/DoubleConverter.cs [hunk 21]
```diff
@@ -16,24 +16,12 @@ public class DoubleConverter : BaseNumberConverter
...
         /// The Type this converter is targeting (e.g. Int16, UInt32, etc.)
         /// </summary>
-        internal override Type TargetType
-        {
-            get
-            {
-                return typeof(Double);
-            }
-        }
+        internal override Type TargetType => typeof(Double);
```
Comment (AlexGhiondea) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/DoubleConverter.cs]: Double -> double

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/EnumConverter.cs [hunk 22]
```diff
@@ -48,13 +49,7 @@ public EnumConverter(Type type)
...
         ///    <para>[To be supplied.]</para>
         /// </summary>
-        protected Type EnumType
-        {
-            get
-            {
-                return _type;
-            }
-        }
+        protected Type EnumType => _type;
```
Comment (AlexGhiondea) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/EnumConverter.cs]: Can we get rid of _type altogether?

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/EnumConverter.cs [hunk 23]
```diff
@@ -16,6 +16,7 @@ namespace System.ComponentModel
     /// </summary>
     public class EnumConverter : TypeConverter
     {
+        private static readonly char[] s_comma = {','};
```
Comment (AlexGhiondea) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/EnumConverter.cs]: Consider renaming this to `s_separators` to convey the fact that this is a separator array.

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/EventDescriptorCollection.cs [hunk 24]
```diff
@@ -39,12 +39,12 @@ public EventDescriptorCollection(EventDescriptor[] events)
             if (events == null)
             {
                 _events = Array.Empty<EventDescriptor>();
-                _eventCount = 0;
+                Count = 0;
```
Comment (AlexGhiondea) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/EventDescriptorCollection.cs]: I think this is not needed.

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/EventDescriptorCollection.cs [hunk 25]
```diff
@@ -380,15 +368,15 @@ protected void InternalSort(string[] names)
 
             if (names != null && names.Length > 0)
             {
-                ArrayList eventArrayList = new ArrayList(_events);
+                List<EventDescriptor> eventArrayList = new List<EventDescriptor>(_events);
```
Comment (AlexGhiondea) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/EventDescriptorCollection.cs]: We can probably rename this now to `eventList` 😄

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Int16Converter.cs [hunk 26]
```diff
@@ -15,13 +15,7 @@ public class Int16Converter : BaseNumberConverter
...
         /// The Type this converter is targeting (e.g. Int16, UInt32, etc.)
         /// </summary>
-        internal override Type TargetType
-        {
-            get
-            {
-                return typeof(Int16);
-            }
-        }
+        internal override Type TargetType => typeof(Int16);
```
Comment (AlexGhiondea) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Int16Converter.cs]: Int16 -> short

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Int32Converter.cs [hunk 27]
```diff
@@ -15,13 +15,7 @@ public class Int32Converter : BaseNumberConverter
...
         /// The Type this converter is targeting (e.g. Int16, UInt32, etc.)
         /// </summary>
-        internal override Type TargetType
-        {
-            get
-            {
-                return typeof(Int32);
-            }
-        }
+        internal override Type TargetType => typeof(Int32);
```
Comment (AlexGhiondea) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Int32Converter.cs]: Int32 -> int

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Int64Converter.cs [hunk 28]
```diff
@@ -15,13 +15,7 @@ public class Int64Converter : BaseNumberConverter
...
         /// The Type this converter is targeting (e.g. Int16, UInt32, etc.)
         /// </summary>
-        internal override Type TargetType
-        {
-            get
-            {
-                return typeof(Int64);
-            }
-        }
+        internal override Type TargetType => typeof(Int64);
```
Comment (AlexGhiondea) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Int64Converter.cs]: Int64 -> long

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/MemberDescriptor.cs [hunk 29]
```diff
@@ -97,19 +97,19 @@ protected MemberDescriptor(MemberDescriptor oldMemberDescriptor, Attribute[] new
             _displayName = oldMemberDescriptor.DisplayName;
             _nameHash = _name.GetHashCode();
 
-            ArrayList newArray = new ArrayList();
+            List<Attribute> newArray = new List<Attribute>();
```
Comment (AlexGhiondea) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/MemberDescriptor.cs]: Consider renaming this to 'newList'

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/PropertyDescriptorCollection.cs [hunk 30]
```diff
@@ -397,15 +385,15 @@ protected void InternalSort(string[] names)
 
             if (names != null && names.Length > 0)
             {
-                ArrayList propArrayList = new ArrayList(_properties);
+                List<PropertyDescriptor> propArrayList = new List<PropertyDescriptor>(_properties);
```
Comment (AlexGhiondea) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/PropertyDescriptorCollection.cs]: Consider renaming this to 'propList'

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/SByteConverter.cs [hunk 31]
```diff
@@ -16,13 +16,7 @@ public class SByteConverter : BaseNumberConverter
...
         /// The Type this converter is targeting (e.g. Int16, UInt32, etc.)
         /// </summary>
-        internal override Type TargetType
-        {
-            get
-            {
-                return typeof(SByte);
-            }
-        }
+        internal override Type TargetType => typeof(SByte);
```
Comment (AlexGhiondea) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/SByteConverter.cs]: SByte -> sbyte

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/SingleConverter.cs [hunk 32]
```diff
@@ -16,24 +16,12 @@ public class SingleConverter : BaseNumberConverter
...
         /// The Type this converter is targeting (e.g. Int16, UInt32, etc.)
         /// </summary>
-        internal override Type TargetType
-        {
-            get
-            {
-                return typeof(Single);
-            }
-        }
+        internal override Type TargetType => typeof(Single);
```
Comment (AlexGhiondea) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/SingleConverter.cs]: Single -> float

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/UInt16Converter.cs [hunk 33]
```diff
@@ -15,13 +15,7 @@ public class UInt16Converter : BaseNumberConverter
...
         /// The Type this converter is targeting (e.g. Int16, UInt32, etc.)
         /// </summary>
-        internal override Type TargetType
-        {
-            get
-            {
-                return typeof(UInt16);
-            }
-        }
+        internal override Type TargetType => typeof(UInt16);
```
Comment (AlexGhiondea) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/UInt16Converter.cs]: Uint16 -> ushort

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/UInt32Converter.cs [hunk 34]
```diff
@@ -15,13 +15,7 @@ public class UInt32Converter : BaseNumberConverter
...
         /// The Type this converter is targeting (e.g. Int16, UInt32, etc.)
         /// </summary>
-        internal override Type TargetType
-        {
-            get
-            {
-                return typeof(UInt32);
-            }
-        }
+        internal override Type TargetType => typeof(UInt32);
```
Comment (AlexGhiondea) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/UInt32Converter.cs]: UInt32- > uint

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/UInt64Converter.cs [hunk 35]
```diff
@@ -15,13 +15,7 @@ public class UInt64Converter : BaseNumberConverter
...
         /// The Type this converter is targeting (e.g. Int16, UInt64, etc.)
         /// </summary>
-        internal override Type TargetType
-        {
-            get
-            {
-                return typeof(UInt64);
-            }
-        }
+        internal override Type TargetType => typeof(UInt64);
```
Comment (AlexGhiondea) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/UInt64Converter.cs]: UInt64 -> ulong

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/CultureInfoConverter.cs (hunk 17 above)
Comment (bmeverett) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/CultureInfoConverter.cs]: I used the example from the issue and you had it removed there. Should it be added back in?

Comment (stephentoub) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/CultureInfoConverter.cs]: With this change, volatile isn't necessary anymore... it's almost the opposite of readonly, where readonly effectively says "this will never change so feel free to cache it as long as you like" and volatile kind of says "this can change at any moment and it's important you use the absolute latest".

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/MaskedTextProvider.cs [hunk 36]
```diff
@@ -668,47 +625,23 @@ public bool IsPassword
...
         ///     A negative value representing an index outside the test string.
         /// </summary>
-        public static int InvalidIndex
-        {
-            get
-            {
-                return invalidIndex;
-            }
-        }
+        public static int InvalidIndex => invalidIndex;
```
Comment (safern) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/MaskedTextProvider.cs]: Should we update all constants from this class to be uppercase? e.g `invalidIndex` to `INVALID_INDEX` ?

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/NestedContainer.cs [hunk 37]
```diff
@@ -126,34 +118,21 @@ private void OnOwnerDisposed(object sender, EventArgs e)
...
 
             // The container in which the component is sited.
-            public IContainer Container
-            {
-                get
-                {
-                    return _container;
-                }
-            }
+            public IContainer Container => _container;
```
Comment (safern) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/NestedContainer.cs]: Can we delete `_container` and just use `Container` ?

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/ToolboxItemFilterAttribute.cs [hunk 38]
```diff
@@ -69,49 +67,27 @@ public ToolboxItemFilterAttribute(string filterString, ToolboxItemFilterType fil
         {
             if (filterString == null) filterString = String.Empty;
```
Comment (safern) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/ToolboxItemFilterAttribute.cs]: Instead of doing
```cs
if (filterString == null) filterString = String.Empty
```
we could simplify this to one line when initializing `FilterString` : 

```cs
FilterString = filterString ?? string.Empty;
```

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/ReflectTypeDescriptionProvider.ReflectedTypeData.cs [hunk 39]
```diff
@@ -173,20 +167,17 @@ internal string GetClassName(object instance)
...
-                        }
-                        else
-                        {
-                            return site.Name;
-                        }
+                        return nestedSite.FullName;
+                    }
+                    else
+                    {
+                        return site.Name;
```
Comment (safern) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/ReflectTypeDescriptionProvider.ReflectedTypeData.cs]: We could simplify this to: 

```cs
return nestedSite != null ? nestedSite.FullName : site.Name;
```
Or I think we could use ?? operator as well:

```cs
return (nestedSite?.FullName) ?? site.Name;
```

**Comment by bmeverett** [comment]
I think the tests may have gotten stuck. Not sure if it was something I did.

**Comment by safern** [comment]
The details of Innerloop CentOS7.1 are not found on the link provided. 
Innerloop OSX Debug has an error, I'm taking a look on that one. 
Innerloop OSX Release actually passed but never updated the status here on the PR.

@mmitche

**Comment by mmitche** [comment]
@safern I beleive this is because the lab is being moved.  Please ignore.

**Comment by safern** [comment]
Ok then I'm going to go ahead and restart both tests with failures. 

test Innerloop CentOS7.1 Debug Build and Test please
test Innerloop OSX Release Build and Test please

**Comment by stephentoub** [comment]
@dotnet-bot test this please

**Comment by bmeverett** [comment]
Thanks guys!
